### PR TITLE
Ed1001 ticket currency select patch

### PR DESCRIPTION
**Pull request by ed1001**
**Checklist**

- [x] I have read the [Contribution & Best practices Guide](https://github.com/openSUSE/osem/blob/master/CONTRIBUTING.md).
- [x] My branch is up-to-date with the upstream `master` branch.
- [ ] The tests pass locally with my changes. (I'm not sure about this as even the untouched project has failures, so I assume 100% pass is not expected? I may need guidance on this one)

**Short description of what this resolves/which [issues](https://github.com/openSUSE/osem/issues) does this fix?:**

- Ticket currencies - #2086

**Changes proposed in this pull request:**

- Ticket price currency select will behave such that if there is already a ticket created, any new tickets will no longer have the option to select a different currency. If there is only one ticket created I have left the option to edit the currency for that ticket (not sure if this is desired, perhaps could disable if sold tickets > 0). I feel like this part of the code could be 'DRYer' but I couldn't seem to make it so, apologies but I have no experience with haml and couldn't find something suitable by googling, if there's a better way I'd be glad to know. Thanks

### CODE REVIEW

File: app/views/admin/tickets/_form.html.haml [hunk 1]
```diff
@@ -12,7 +12,10 @@
       = f.input :title, input_html: { autofocus: true }
       = f.input :description, input_html: { rows: 5, data: { provide: 'markdown-editable' } }
       = f.input :price
-      = f.input :price_currency, as: :select, class: 'form-control', collection: ['USD', 'EUR', 'GBP', 'INR', 'CNY', 'CHF'], include_blank: false
+      - if (@conference.tickets.count >= 1 && @ticket.new_record?) || @conference.tickets.count > 1
+        = f.input :price_currency, as: :select, class: 'form-control', collection: [@conference.tickets.first.price_currency], include_blank: false
```
Comment (Ana06) [inline app/views/admin/tickets/_form.html.haml]: this shouldn't be a select with a collection, as you only have one. It should be just shown.

File: app/views/admin/tickets/_form.html.haml [hunk 2]
```diff
@@ -12,7 +12,10 @@
       = f.input :title, input_html: { autofocus: true }
       = f.input :description, input_html: { rows: 5, data: { provide: 'markdown-editable' } }
       = f.input :price
-      = f.input :price_currency, as: :select, class: 'form-control', collection: ['USD', 'EUR', 'GBP', 'INR', 'CNY', 'CHF'], include_blank: false
+      - if (@conference.tickets.count >= 1 && @ticket.new_record?) || @conference.tickets.count > 1
```
Comment (Ana06) [inline app/views/admin/tickets/_form.html.haml]: I am not sure if I follow this condition, why isn't it enough with `@conference.tickets.count > 1` `count` raises a query to the database so new records doesn't show up.

Comment (ed1001) [reply app/views/admin/tickets/_form.html.haml]: With only @conference.tickets.count > 1 the second added ticket would still be able to change the currency. @conference.tickets.count >= 1 && @ticket.new_record? makes sure that you can still change currency if you're editing the first ticket (I went on the assumption that was desired, but now I see it isn't).

File: app/views/admin/tickets/_form.html.haml (hunk 1 above)
Comment (ed1001) [reply app/views/admin/tickets/_form.html.haml]: I had trouble with form submission without keeping the input. Thinking about it now I guess there's probably an easy way to add a read only input.

**Comment by ed1001** [comment]
Hi @Ana06 ! 

Thanks a lot for the feedback :)

I'm certainly up for the more extensive changes you proposed. As this is a slightly larger task I'd like to run a couple of things by you if you don't mind.

 - Remove line 11 (monetize) from ticket.rb and add to conference.rb
 - Add price_currency field to conferences and remove from tickets table
 - change all refs to currency from 'ticket.price_currency' => 'ticket.conference.price_currency'
 - change in views (conference and ticket forms)

With regards to your 3rd suggestion:

>  3. create a data rake task to set this new attribute to the tickets currency (it should be the same in all the ticket but I would double check this in the rake task - also don't forget to document this rake task for the updates)

I don't understand why this is necessary, or how even possible if the price_currency is removed from ticket. If the price_currency is always referenced through conference then the conference's tickets must always share that price_currency. Apologies for my naivety, I'm still finding my feet with Rails :)

**Comment by ed1001** [comment]
Actually maybe best to keep the price_currency in tickets table and have the conference's currency attribute dictate the tickets' currency at creation. Then can keep things like `humanized_money_with_symbol ticket.price`
in the views

**Comment by ed1001** [comment]
Hi @Ana06 ! 

Thanks a lot for the feedback :)

I'm certainly up for the more extensive changes you proposed. As this is a slightly larger task I'd like to run a couple of things by you if you don't mind. 

 - Just to make sure, by `4. remove the currency attribute from the ticket` I guess you mean remove it from the view, not remove the attribute from the ticket model, is that correct?
 - Is the data rake task primarily for checking? I assume the ticket's currency will be set at creation via tickets_controller.
 - As for documenting the rake task, would the details go in README_FOR_APP?

**Comment by Ana06** [comment]
@ed1001 

> Just to make sure, by 4. remove the currency attribute from the ticket I guess you mean remove it from the view, not remove the attribute from the ticket model, is that correct?

No, I mean removing it from the model, as it is already in the conference. It should be still rendered in the view (but you can do this as you have it in the conference).

> Is the data rake task primarily for checking? I assume the ticket's currency will be set at creation via tickets_controller.

The rake task is to set the new attribute in conference for old conferences, as it is empty for new conferences. But as we have it in the first ticket, we can just copy over this information to conference, before getting rid of the the info in the ticket.

> As for documenting the rake task, would the details go in README_FOR_APP?

In https://github.com/openSUSE/osem/blob/master/CHANGES.md as this is only for updating old instances. New instances don't need this. :wink: 

As this is a bigger feature, I recommend you made independent commits for the steps I proposed (step 4 could be done in a different PR as well, so that it doesn't get too big). And of course, If you have doubts/help, do not hesitate to ask :wink:

**Comment by ed1001** [comment]
Thanks for the clarification @Ana06, I will get on it!

**Pull request closed without merging**